src/components/ImageLightbox/component.js

```diff
@@ -0,0 +1,176 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { View, Platform, TouchableOpacity, Modal, ActivityIndicator, Share, Linking, Image } from 'react-native';
+import FastImage from 'react-native-fast-image';
+import ImageViewer from 'react-native-image-zoom-viewer';
+import Icon from 'react-native-vector-icons/Ionicons';
+import { isDataUrl } from 'kitsu/common/utils/url';
+import { isEmpty } from 'lodash';
+import { styles } from './styles';
+
+export class ImageLightbox extends PureComponent {
+  static propTypes = {
+    images: PropTypes.arrayOf(PropTypes.string).isRequired,
+    visible: PropTypes.bool,
+    initialImageIndex: PropTypes.number,
+    onClose: PropTypes.func.isRequired,
+    onShare: PropTypes.func,
+    onDownload: PropTypes.func,
+    onOpen: PropTypes.func,
+  }
+
+  static defaultProps = {
+    visible: false,
+    initialImageIndex: 0,
+    onShare: null,
+    onDownload: null,
+    onOpen: null,
+  }
+
+  shareImage = (url) => {
+    if (isEmpty(url)) return;
+    Share.share({ url });
+  }
+
+  downloadImage = (url) => {
+    console.log(url);
+  }
+
+  openImage = async (url) => {
+    if (isEmpty(url)) return;
+
+    try {
+      const supported = await Linking.canOpenURL(url);
+      if (supported) {
+        Linking.openURL(url);
+      }
+    } catch (e) {
+      console.log(`Error handling ${url}: ${e}`);
+    }
+  }
+
+  renderFooter(imageUrls) {
+    const { onClose, onShare, onDownload, onOpen } = this.props;
+    const shareImage = onShare || this.shareImage;
+    const downloadImage = onDownload || this.downloadImage;
+    const openImage = onOpen || this.openImage;
+
+    return (currentIndex) => {
+      const currentImage = imageUrls[currentIndex];
+      const url = typeof currentImage === 'string' ? currentImage : (currentImage && currentImage.url) || null;
+      const isRemoteUrl = url && url.includes('http');
```

Comment:
Could this ever flag true if `http` exists in the string outside of the protocol?